backup/backup-disks.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python3
+
+""" Primary Backup Logic"""
+import json
+import sys
+import datetime
+
+from datetime import date
+from settings import settings
+from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
+from json.decoder import JSONDecodeError as JsonError
+
+
+def list_disks(compute, project, zone):
+    """ Lists all persistent disks used by project """
+    all_disks = []
+    result = compute.disks().list(project=project, zone=zone).execute()
+    all_disks.extend(result['items'])
+
+    while 'nextPageToken' in result:
+        result = compute.disks().list(project=project, zone=zone, \
+            pageToken=result['nextPageToken']).execute()
+        all_disks.extend(result['items'])
+
+    return all_disks
+
+
+def list_snapshots(compute, project):
+    """ Lists all snapshots created for this project """
+    all_snapshots = []
+    result = compute.snapshots().list(project=project).execute()
+    all_snapshots.extend(result['items'])
+
+    while 'nextPageToken' in result:
+        result = compute.snapshots().list(project=project, \
+            pageToken=result['nextPageToken']).execute()
+        all_snapshots.extend(result['items'])
+
+    return all_snapshots
+
+
+def filter_disks_by_name(disks, name):
+    """ Takes in NAME, a predefined prefix to disk names, and filters 
+    disks to only returns those that have NAME """
+    filtered_disks = []
+    for disk in disks:
+        try:
+            if name in disk['name']:
+                filtered_disks.append(disk)
+        except KeyError:
+            sys.exit("Improperly formatted disks -- is your information correct?")
+    return filtered_disks
+
+
+def filter_disks_by_namespace(disks, namespace, namespace_dict_key):
+    """ Takes in NAMESPACE, a predefined string value, and filters
+    disks to only return those belong to NAMESPACE """
+    filtered_disks = []
+    for disk in disks:
+        try:
+            disk_namespace = json.loads(disk['description'])[namespace_dict_key]
+            if disk_namespace == namespace:
+                filtered_disks.append(disk)
+        except (JsonError, KeyError):
+            continue
+    return filtered_disks
+
+
+def filter_snapshots_by_time(snapshots, retention_period):
+    """ Takes in RETENTION_PERIOD, a number of days, and filters
+    disks to return only those that are older than the retention_period
+    and should be deleted """
+    try:
+        old_snapshots = list(filter(lambda snapshot: \
+            __days_between_now_and_last_backup(snapshot['creationTimestamp'][:10]) > retention_period, snapshots))
```

Comment:
I'm guessing the date information is within the first `10` chars of the `creationTimestamp`. Would this fit better in an constant var to improve clairty?